contracts/plugins/posts/lib.rs

```diff
@@ -337,6 +354,91 @@ mod posts {
       Ok(())
     }
 
+    #[ink(message)]
+    #[modifiers(only_active_member)]
+    pub fn new_comment(&mut self, post_id: PostId, content: String) -> PostResult<CommentId> {
+      let author = self.env().caller();
+
+      if let Some(_) = self.posts.get(post_id) {
+        let new_comment_id = self.comments_nonce.get_or_default();
+        let next_comment_nonce = new_comment_id.saturating_add(1);
+
+        let comment = Comment {
+          post_id,
+          content,
+          author,
+          created_at: Self::env().block_timestamp(),
+          updated_at: None
+        };
+
+        self.comments.insert(new_comment_id, &comment);
+
+        let mut comments = self.post_to_comments.get(post_id).unwrap_or(Vec::new());
+        comments.push(new_comment_id);
+        self.post_to_comments.insert(post_id, &comments);
+
+        self.comments_nonce.set(&next_comment_nonce);
+
+        Ok(new_comment_id)
+      } else {
+        Err(PostError::PostNotExisted)
+      }
```

Comment:
We should do an early check and return for this case, this helps to avoid a long nested block like this!